src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java

```diff
@@ -0,0 +1,442 @@
+    CdfGcsActions.getSchema();
+    SeleniumHelper.waitElementIsVisible(CdfGCSLocators.getSchemaLoadComplete, 10L);
+    SeleniumHelper.waitElementIsVisible(GCSLocators.outputSchemaColumnNames.get(0), 2L);
+    int index = 0;
+    for (WebElement element : GCSLocators.outputSchemaColumnNames) {
+      propertiesSchemaColumnList.add(element.getAttribute("value"));
+      sourcePropertiesOutputSchema.put(element.getAttribute("value"),
+                                       GCSLocators.outputSchemaDataTypes.get(index).getAttribute("title"));
+      index++;
+    }
+    Assert.assertTrue(propertiesSchemaColumnList.size() >= 1);
+  }
+
+  @Then("Capture and validate output schema for blob format")
+  public void captureAndValidateOutputSchemaForBlobFormat() {
+    CdfGcsActions.getSchema();
+    SeleniumHelper.waitElementIsVisible(CdfGCSLocators.getSchemaLoadComplete, 10L);
+    SeleniumHelper.waitElementIsVisible(GCSLocators.outputSchemaColumnNames.get(0), 2L);
+    int index = 0;
+    for (WebElement element : GCSLocators.outputSchemaColumnNames) {
+      propertiesSchemaColumnList.add(element.getAttribute("value"));
+      sourcePropertiesOutputSchema.put(element.getAttribute("value"),
+                                       GCSLocators.outputSchemaDataTypes.get(index).getAttribute("title"));
+      index++;
+    }
+    Assert.assertTrue(propertiesSchemaColumnList.size() == 1);
+  }
+
+  @Then("Validate GCS properties")
+  public void validateBigqueryProperties() {
+    CdfGcsActions.clickValidateButton();
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.pluginValidationSuccessMsg, 10L);
+    String expectedErrorMessage = E2ETestUtils.errorProp(E2ETestConstants.ERROR_MSG_VALIDATION);
+    String actualErrorMessage = CdfStudioLocators.pluginValidationSuccessMsg.getText();
+    Assert.assertEquals(expectedErrorMessage, actualErrorMessage);
+  }
+
+  @Then("Close the GCS Properties")
+  public void closeTheGCSProperties() {
+    CdfGcsActions.closeButton();
+  }
+
+  @Then("Open BigQuery Properties")
+  public void openBigQueryProperties() {
+    CdfStudioActions.clickProperties("BigQuery");
+  }
+
+  @Then("Enter the BigQuery Sink properties for table {string}")
+  public void enterTheBigQueryPropertiesForTable(String tableName) throws InterruptedException, IOException {
+    CdfBigQueryPropertiesActions.enterProjectId(E2ETestUtils.pluginProp("projectId"));
+    CdfBigQueryPropertiesActions.enterDatasetProjectId(E2ETestUtils.pluginProp("projectId"));
+    CdfBigQueryPropertiesActions.enterBigQueryReferenceName("BQ_Ref_" + UUID.randomUUID().toString());
+    CdfBigQueryPropertiesActions.enterBigQueryDataset(E2ETestUtils.pluginProp("dataset"));
+    CdfBigQueryPropertiesActions.enterBigQueryTable(E2ETestUtils.pluginProp(tableName));
+    CdfBigQueryPropertiesActions.clickUpdateTable();
+    CdfBigQueryPropertiesActions.clickTruncatableSwitch();
+  }
+
+  @Then("Validate BigQuery properties")
+  public void validateBigQueryProperties() {
```

Comment:
Method at Line #182 seems similar to this method. Can we have a common method and get it called from both?

Reply:
Created common method